test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +172,72 @@ describe('View existing record,', function() {
+            chaisePage.waitForElement(fiddlerBtn);
+        });
+        it('Table of contents should be displayed by default', function(){
+            var recordSidePan = chaisePage.recordPage.getSidePanel();
+            expect(recordSidePan.isDisplayed()).toBeTruthy("Side Panel is not visible when page loads initially.");
+        });
+
+        it('On click of Related table name in TOC, page should move to the contents and open the table details', function(){
+            var rtTableHeading = chaisePage.recordPage.getRelatedTableHeading(testParams.sidePanelTest.tableToShow);
+
+            recSidePanelCat_5.click().then(function(className) {
+              // related table should be visible
+                expect(rtTableHeading.isDisplayed()).toBeTruthy("Category_5 heading is not visible.");
+                return rtTableHeading.getAttribute("class");
+            }).then (function(className) {
+                expect(className).toContain("panel-open", "Related table panel is not open when clicked through TOC.");
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
+            });
+        });
+
+        it('Record count along with heading should match for the panel and related table content should be in correct order', function(){
+            chaisePage.recordPage.getSidePanelTableTitles().then(function(tableNames){
+              for (var i=0; i<tableNames.length; i++){
+                tableNames[i] = tableNames[i].replace(/ +/g, " ");
+              }
+                expect(tableNames).toEqual(testParams.sidePanelTest.sidePanelTableOrder,"Order is not maintained for related tables in side panel");
+                expect(tableNames.length).toEqual(testParams.sidePanelTest.tocCount, "Count mismatch for number of related tables in side panel");
+                return chaisePage.recordPage.getSidePanelHeading();
+            }).then(function(sidePanelHeading){
+                expect(sidePanelHeading).toBe(testParams.sidePanelTest.panelHeading, "Side Panel heading did not match.");
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
+            });
+        });
+
+        it('Side panel should hide/show by clicking pull button', function(){
+            var recPan =  chaisePage.recordPage.getSidePanel();
+
+            fiddlerBtn.getAttribute("class").then(function(classNameRight) {
+                expect(classNameRight).toContain('glyphicon glyphicon-triangle-right', 'Side Pan Pull button is not pointing in the right direction');
+                return recPan.getAttribute("class");
+            }).then(function(openPanel){
+                expect(openPanel).toContain('open-panel', 'Side Panel is not visible when fiddler is poining in right direction');
+                fiddlerBtn.click();
```

Comment:
You need to return the promise (click event) here and chain it to make sure that you have clicked on the button and then check for the attribute.

Also I think you can change all these getAttribute test cases in just one statement. `expect(recPan.getAttribute("class")).toContain(CLASS, ERR_MESSAGE)`